runtime/pkg/rilltime/rilltime_test.go

```diff
@@ -1,176 +1,576 @@
+		{"2025-03-23T00:00:00Z", "2025-03-30T00:00:00Z", "2025-04-06T00:00:00Z"},
+		{"2024-12-22T00:00:00Z", "2024-12-29T00:00:00Z", "2025-01-05T00:00:00Z"},
+		{"2025-04-27T00:00:00Z", "2025-05-04T00:00:00Z", "2025-05-11T00:00:00Z"},
+		{"2024-10-27T00:00:00Z", "2024-11-03T00:00:00Z", "2024-11-10T00:00:00Z"},
+		{"2025-02-23T00:00:00Z", "2025-03-02T00:00:00Z", "2025-03-09T00:00:00Z"},
+		{"2024-11-24T00:00:00Z", "2024-12-01T00:00:00Z", "2024-12-08T00:00:00Z"},
+	}
+
+	var testCases []testCase
+
+	for i, weekday := range monthsForWeekdays {
+		testCases = append(testCases, testCase{
+			timeRange: fmt.Sprintf("W1 as of %s", weekday),
+			start:     weekBoundaries[i][1],
+			end:       weekBoundaries[i][2],
+			grain:     timeutil.TimeGrainDay,
+		})
+		testCases = append(testCases, testCase{
+			timeRange: fmt.Sprintf("<1W as of %s", weekday),
+			start:     weekBoundaries[i][1],
+			end:       weekBoundaries[i][2],
+			grain:     timeutil.TimeGrainDay,
+		})
+		testCases = append(testCases, testCase{
+			timeRange: fmt.Sprintf(">1W of -1M! as of %s", weekday),
+			start:     weekBoundaries[i][0],
+			end:       weekBoundaries[i][1],
+			grain:     timeutil.TimeGrainDay,
+		})
+
+		testCases = append(testCases, testCase{
+			timeRange: fmt.Sprintf("W1 as of %s", weekday),
+			start:     weekBoundariesForSunday[i][1],
+			end:       weekBoundariesForSunday[i][2],
+			grain:     timeutil.TimeGrainDay,
+			FirstDay:  7,
+		})
+		testCases = append(testCases, testCase{
+			timeRange: fmt.Sprintf("<1W as of %s", weekday),
+			start:     weekBoundariesForSunday[i][1],
+			end:       weekBoundariesForSunday[i][2],
+			grain:     timeutil.TimeGrainDay,
+			FirstDay:  7,
+		})
+		testCases = append(testCases, testCase{
+			timeRange: fmt.Sprintf(">1W of -1M! as of %s", weekday),
+			start:     weekBoundariesForSunday[i][0],
+			end:       weekBoundariesForSunday[i][1],
+			grain:     timeutil.TimeGrainDay,
+			FirstDay:  7,
+		})
+	}
+
+	runTests(t, testCases, now, minTime, maxTime, watermark, nil)
+}
+
+func Test_IsoTimeRanges(t *testing.T) {
+	testCases := []testCase{
+		{"2025-02-20T01:23:45Z to 2025-07-15T02:34:50Z", "2025-02-20T01:23:45Z", "2025-07-15T02:34:50Z", timeutil.TimeGrainSecond, 1, 1},
+		{"2025-02-20T01:23:45Z / 2025-07-15T02:34:50Z", "2025-02-20T01:23:45Z", "2025-07-15T02:34:50Z", timeutil.TimeGrainSecond, 1, 1},
+
+		{"2025-02-20T01:23", "2025-02-20T01:23:00Z", "2025-02-20T01:24:00Z", timeutil.TimeGrainSecond, 1, 1},
+		{"2025-02-20T01", "2025-02-20T01:00:00Z", "2025-02-20T02:00:00Z", timeutil.TimeGrainMinute, 1, 1},
+		{"2025-02-20", "2025-02-20T00:00:00Z", "2025-02-21T00:00:00Z", timeutil.TimeGrainHour, 1, 1},
+		{"2025-02", "2025-02-01T00:00:00Z", "2025-03-01T00:00:00Z", timeutil.TimeGrainDay, 1, 1},
+		{"2025", "2025-01-01T00:00:00Z", "2026-01-01T00:00:00Z", timeutil.TimeGrainMonth, 1, 1},
+	}
+
+	runTests(t, testCases, now, minTime, maxTime, watermark, nil)
 }
 
 func Test_Eval_watermark_on_boundary(t *testing.T) {
```

Comment:
Test function names should use pascal case, optionally using underscores for namespacing, e.g. `TestEval_WatermarkOnBoundary`